src/kseq_load.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * @file kseq_load.cpp
+ * @brief Load input reads directly into RAM (FASTA, FASTQ and compressed formats)
+ * @parblock
+ * SortMeRNA - next-generation reads filter for metatranscriptomic or total RNA
+ * @copyright 2012-16 Bonsai Bioinformatics Research Group
+ * @copyright 2014-16 Knight Lab, Department of Pediatrics, UCSD, La Jolla
+ *
+ * This file is part of SortMeRNA.
+ *
+ * SortMeRNA is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * SortMeRNA is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with SortMeRNA.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @authors jenya.kopylov@gmail.com
+ *          laurent.noe@lifl.fr
+ *          helene.touzet@lifl.fr
+ *          pierre.pericard@lifl.fr
+ *          mikael.salson@lifl.fr
+ *          robknight@ucsd.edu
+ */
+
+#include "../include/kseq_load.hpp"
+
+
+/*
+ *
+ * FUNCTION : load_reads()
+ * PURPOSE  : load reads using generic buffer
+ * OUTPUT   : double pointer array to sequences in buffer
+ * See complete documentation in include/mmap.hpp
+ *
+ *******************************************************/
+char**
+load_reads(char* inputreads,
+           char* raw,
```

Comment:
on input, is `raw` expected to be `NULL`? If so, would it make sense to assert that to avoid a potential memory leak?